### PR TITLE
feat(@vtmn/svelte): add `VtmnToast` component

### PR DESCRIPTION
**Pull request by Tlahey**
Create a `VtmnToast` component for svelte.

How it works : 
- `VtmnToast` need to be put on the main body of the application.
- `VtmnToastItem` are used by the `VtmnToast` to display toasts
- A store `vtmnToastStore` can be imported everywhere on the application, you can use the function `vtmnToastStore.send({ content: 'Hello world!' })` to send toast to display

image

### CODE REVIEW

File: packages/showcases/core/csf/components/overlays/toast.csf.js [hunk 1]
```diff
@@ -7,3 +7,28 @@ export const parameters = {
     url: 'https://www.figma.com/file/zDZIyayUlr1yTWrsi7cFoo/?node-id=2790%3A10831',
   },
 };
+
+export const argTypes = {
+  content: {
+    type: { name: 'string', required: true },
+    description: 'Text display on the toast',
```
Comment (Tlahey) [inline packages/showcases/core/csf/components/overlays/toast.csf.js:14]: displayed

File: packages/sources/svelte/src/components/overlays/VtmnToast/VtmnToastItem.svelte [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+   */
+  export let withIcon = false;
+
+  /**
+   * @type {string} content text of the toast
+   */
+  export let content;
+
+  /**
+   * @type {string} display a close button on right
```
Comment (Tlahey) [inline packages/sources/svelte/src/components/overlays/VtmnToast/VtmnToastItem.svelte:16]: Just `display a close button`

File: packages/sources/svelte/src/components/overlays/VtmnToast/VtmnToastItem.svelte [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+   */
+  export let withCloseButton = false;
+
+  /**
+   * @type {number} timeout before the component execute the close action
```
Comment (Tlahey) [inline packages/sources/svelte/src/components/overlays/VtmnToast/VtmnToastItem.svelte:21]: `Time before the component dispatch automatically 'close'`

File: packages/sources/svelte/src/components/overlays/VtmnToast/vtmnToastStore.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+class VtmnToastStore {
+  constructor() {
+    this._toasts = writable([]);
+    this._id = 0;
+  }
+
+  get newId() {
+    this._id++;
+    return this._id;
```
Comment (Tlahey) [inline packages/sources/svelte/src/components/overlays/VtmnToast/vtmnToastStore.js:11]: `return ++this._id`

File: packages/sources/svelte/src/components/overlays/VtmnToast/README.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+```
+npm install @vtmn/svelte
+```
+
+Now, you can use `VtmnToast` on your app.
+
+# How it works
+
+First, you need to place the `VtmnToast` component preferably at your
```
Comment (Tlahey) [inline packages/sources/svelte/src/components/overlays/VtmnToast/README.md:13]: First, you need to place the `VtmnToast` component on the top of your application

File: packages/sources/svelte/src/components/overlays/VtmnToast/README.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
+# Quick start
+
+You need to add `@vtmn/svelte` on your `package.json`
```
Comment (Tlahey) [inline packages/sources/svelte/src/components/overlays/VtmnToast/README.md:3]: + `@vtmn/icon`

File: packages/sources/svelte/src/components/overlays/VtmnToast/test/vtmnToastItem.spec.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  };
+
+  test("Should be visible with a class 'vtmn-toast' and 'show'", () => {
+    const { container } = render(VtmnToastItem, {
+      content: 'Unit-test Toast',
+      timeout,
+    });
+    expect(getToast(container)).toHaveClass('vtmn-toast', 'show');
+  });
+  test('Should display the content of the toast', () => {
```
Comment (Tlahey) [inline packages/sources/svelte/src/components/overlays/VtmnToast/test/vtmnToastItem.spec.js:32]: Add test `Should apply custom class to toast`

File: packages/sources/svelte/src/components/overlays/VtmnToast/VtmnToastItem.svelte [hunk 8]
```diff
@@ -0,0 +1,57 @@
+<script>
+  import { VtmnButton } from '../../..';
+  import { createEventDispatcher, onDestroy, onMount } from 'svelte';
+
+  /**
+   * @type {boolean} display the toast with a left icon
+   */
```
Comment (Tlahey) [inline packages/sources/svelte/src/components/overlays/VtmnToast/VtmnToastItem.svelte:7]: @defaultValue 'false'

File: packages/sources/svelte/src/components/overlays/VtmnToast/VtmnToastItem.svelte [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  export let withIcon = false;
+
+  /**
+   * @type {string} content text of the toast
+   */
+  export let content;
+
+  /**
+   * @type {string} display a close button on right
+   */
```
Comment (Tlahey) [inline packages/sources/svelte/src/components/overlays/VtmnToast/VtmnToastItem.svelte:17]: @defaultValue 'false'

File: packages/sources/svelte/src/components/overlays/VtmnToast/VtmnToastItem.svelte [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+  let className = '';
+  /**
+   * @type {string} Custom classes to apply to the component.
+   */
+  export { className as class };
+
+  let timeoutId;
+
+  const dispatch = createEventDispatcher();
+  const closeHandler = () => {
```
Comment (Tlahey) [inline packages/sources/svelte/src/components/overlays/VtmnToast/VtmnToastItem.svelte:37]:
```
const closeHandler = () => dispatch('close');
```

File: packages/sources/svelte/src/components/overlays/VtmnToast/test/vtmnToastItem.spec.js [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+
+const timeout = 5000;
+
+describe('VtmnToastItem', () => {
+  const getToast = (container) =>
+    container.getElementsByClassName('vtmn-toast')[0];
+  const getCloseButton = (container) =>
+    container.getElementsByClassName('vtmn-btn')[0];
+
+  const expectedCancelOnElement = async (
```
Comment (Tlahey) [inline packages/sources/svelte/src/components/overlays/VtmnToast/test/vtmnToastItem.spec.js:14]: `expectedCloseOnElement`

**Review by lauthieb — CHANGES_REQUESTED** [review]
Thanks @Tlahey for this contribution.

Do you please know why the toast is not centered? 

CleanShot 2022-03-15 at 11 47 02

File: packages/sources/svelte/src/components/overlays/VtmnToast/test/vtmnToastItem.spec.js (hunk 11 above)
Comment (lauthieb) [reply packages/sources/svelte/src/components/overlays/VtmnToast/test/vtmnToastItem.spec.js:14]: @Tlahey ?

**Review by lauthieb — CHANGES_REQUESTED** [review]

**Review by lauthieb — CHANGES_REQUESTED** [review]
Thanks @Tlahey for your contribution.

Do you know why the toast is not centered please? 

Thanks!

CleanShot 2022-03-15 at 18 43 39

**Comment by Tlahey** [comment]
> Thanks @Tlahey for your contribution.
> 
> Do you know why the toast is not centered please?
> 
> Thanks!
> 
> CleanShot 2022-03-15 at 18 43 39

The toast is centered on all the page, not the preview (class show)

image

On this case, if we remove the right:50%, the toast are correctly align center into the container.

**Comment by Tlahey** [comment]
It's ok ! I don't know why but the `$$restProp` have been deleted. My bad, it's fixed :)

**Review by lauthieb — APPROVED** [review]
Thanks for you work @Tlahey 💪

**Pull request merged**